src/content/content_encoding.rs

```diff
@@ -0,0 +1,147 @@
+//! Specify the compression algorithm.
+
+use crate::content::{Encoding, EncodingProposal};
+use crate::headers::{HeaderName, HeaderValue, Headers, ToHeaderValues, CONTENT_ENCODING};
+
+use std::fmt::{self, Debug};
+use std::ops::{Deref, DerefMut};
+use std::option;
+
+/// Specify the compression algorithm.
+///
+/// # Specifications
+///
+/// - [RFC 7231, section 3.1.2.2: Content-Encoding](https://tools.ietf.org/html/rfc7231#section-3.1.2.2)
+///
+/// # Examples
+///
+/// ```
+/// # fn main() -> http_types::Result<()> {
+/// #
+/// use http_types::Response;
+/// use http_types::content::{ContentEncoding, Encoding};
+/// let mut encoding = ContentEncoding::new(Encoding::Gzip);
+///
+/// let mut res = Response::new(200);
+/// encoding.apply(&mut res);
+///
+/// let encoding = ContentEncoding::from_headers(res)?.unwrap();
+/// assert_eq!(encoding, &Encoding::Gzip);
+/// #
+/// # Ok(()) }
+/// ```
+pub struct ContentEncoding {
+    inner: Encoding,
+}
+
+impl ContentEncoding {
+    /// Create a new instance of `CacheControl`.
+    pub fn new(encoding: Encoding) -> Self {
+        Self { inner: encoding }
```

Comment:
Additionally there should probably be a way to get the `Encoding` out from here.